src/kOS/Module/kOSCustomParameters.cs

```diff
@@ -0,0 +1,187 @@
+            "of an external file. Telnet settings are still stored in the external file. It appears that you still " +
+            "have valid settings stored in the external file, but that this is either a new save or the save has " +
+            "not yet been migrated.  Would you like to migrate now?";
+
+        [GameParameters.CustomParameterUI("")]
+        public bool migrated = false;
+
+        [GameParameters.CustomIntParameterUI("")]
+        public int version = 0;
+
+        [GameParameters.CustomIntParameterUI("Instructions per update", minValue = 50, maxValue = 2000)]
+        public int instructionsPerUpdate = 200;
+
+        [GameParameters.CustomParameterUI("Enable compressed storage")]
+        public bool useCompressedPersistence = true;
+
+        [GameParameters.CustomParameterUI("Show statistics")]
+        public bool showStatistics = false;
+
+        [GameParameters.CustomParameterUI("Enable Remote Tech integration")]
+        public bool enableRTIntegration = true;
+
+        [GameParameters.CustomParameterUI("Start on the archive")]
+        public bool startOnArchive = false;
+
+        [GameParameters.CustomParameterUI("Obey hide UI toggle")]
+        public bool obeyHideUi = true;
+
+        [GameParameters.CustomParameterUI("Enable safe mode")]
+        public bool enableSafeMode = true;
+
+        [GameParameters.CustomParameterUI("Audible exceptions")]
+        public bool audibleExceptions = true;
+
+        [GameParameters.CustomParameterUI("Verbose exceptions")]
+        public bool verboseExceptions = true;
+
+        [GameParameters.CustomParameterUI("Only use Blizzy toolbar")]
+        public bool useBlizzyToolbarOnly = false;
+
+        [GameParameters.CustomParameterUI("Debug each opcode")]
+        public bool debugEachOpcode = false;
+
+        public override GameParameters.GameMode GameMode
+        {
+            get
+            {
+                return GameParameters.GameMode.ANY;
+            }
+        }
+
+        public override bool HasPresets
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override string Section
+        {
+            get
+            {
+                return "kOS";
+            }
+        }
+
+        public override int SectionOrder
+        {
+            get
+            {
+                return 0;
+            }
+        }
+
+        public override string Title
+        {
+            get
+            {
+                return "CONFIG";
+            }
+        }
+
+        public override void OnLoad(ConfigNode node)
+        {
+            base.OnLoad(node);
+            instance = null;
+        }
+
+        public override bool Enabled(MemberInfo member, GameParameters parameters)
+        {
+            if (member.Name == "migrated" || member.Name == "version")
+            {
+                return false;
+            }
+            return base.Enabled(member, parameters);
+        }
```

Comment:
grrrr. dangit github.  I deleted the above comment - I see how it works now.  But github kept the comment in its review anyway.  Ignore it.